core/src/main/java/tc/oc/pgm/filters/FilterParser.java

```diff
@@ -453,4 +457,46 @@ public TimeFilter parseTimeFilter(Element el) throws InvalidXMLException {
   public ScoreFilter parseScoreFilter(Element el) throws InvalidXMLException {
     return new ScoreFilter(XMLUtils.parseNumericRange(new Node(el), Integer.class));
   }
+
+  // Methods for parsing QueryModifiers
+
+  @MethodParser("offset")
+  public LocationQueryModifier parseOffsetFilter(Element el) throws InvalidXMLException {
+    String value = el.getAttributeValue("location");
+    // Check vector format
+    XMLUtils.parseVector(new Node(el), value.replaceAll("[\\^~]", ""));
+
+    String[] coords = value.split("\\s*,\\s*");
+
+    Boolean local = null;
+    for (String coord : coords) {
+      if (local == null) {
+        local = coord.startsWith("^");
+      }
+
+      if (coord.startsWith("^") != local)
+        throw new InvalidXMLException("Cannot mix world & local coordinates", el);
+    }
+
+    if (local == null) throw new InvalidXMLException("No coordinates provided", el);
+
+    if (local) {
+      double x = Double.parseDouble(coords[0].substring(1));
+      double y = Double.parseDouble(coords[1].substring(1));
+      double z = Double.parseDouble(coords[2].substring(1));
+      return new LocalLocationQueryModifier(parseChild(el), new Vector(x, y, z));
```

Comment:
You're already running an XMLUtils.parseVector all the way at the top, no need to parse individual coords in substrings

Reply:
Right, forgot about that